### PR TITLE
Miscellaneous doc updates for plugins

### PR DESCRIPTION
**Pull request by jasonblais**

### CODE REVIEW

File: source/administration/plugins.rst [hunk 1]
```diff
@@ -4,7 +4,7 @@ Plugins (Beta)
 .. note::
   This is the admin documentation for plugins. If you're a developer looking to build a plugin, see `our developer documentation <https://developers.mattermost.com/extend/plugins>`_.
 
-Mattermost supports plugins, which allow users to customize and extend the platform by adding specific features.
+Mattermost supports plugins, which allow users to customize and extend the platform by adding specific features. See our `demo plugin <https://github.com/mattermost/mattermost-plugin-demo>`_ that demonstrates the capabilities of a Mattermost plugin. For a starting point to write a Mattermost plugin, see our `sample plugin <https://github.com/mattermost/mattermost-plugin-demo>`_.
```
Comment (amyblais) [inline source/administration/plugins.rst]: These 2 links are the same, is this expected or can the sentences be combined?

Comment (jasonblais) [reply source/administration/plugins.rst]: Good catch, fixed

File: source/administration/plugins.rst [hunk 2]
```diff
@@ -19,6 +19,8 @@ There are two main types of plugins, both of which are supported in Team and Ent
 
 Later in upcoming releases, the `Enterprise Edition <https://about.mattermost.com/pricing>`_ will also support plugins that enable you to extend Mattermost functionality to meet a specific, complex requirement such as profiling performance metrics, and to implement highly customized compliance rules for `information barriers <http://www.17a-4.com/supervision-information-barriers/>`_.
 
+For a list of available plugins, including those for auto-linking, auto-censoring and anti-virus scanning, as well as third-party plugins with JIRA, Zoom, GitHub and more, see the `Mattermost plugins repo <https://github.com/mattermost/mattermost-plugins>`_.
```
Comment (wiersgallak) [inline source/administration/plugins.rst]: Do we want to link them to the plugins repo and the apps available from the Webpage (https://about.mattermost.com/community-applications/)

Comment (jasonblais) [reply source/administration/plugins.rst]: For now just the plugins repo - the apps available on the Webpage are largely for webhooks/slash commands. We link to the Webpage on those integration docs, e.g. https://docs.mattermost.com/developer/webhooks-incoming.html#share-your-integration

Eventually all the plugins and webhook integrations should be in one place hopefully with the new app directory you're working on :)

**Pull request merged**